### ISSUE TITLE
🐛 Vue `v-on` directives (and shorthand) false positive with `noUnusedVariables` on `next`

### ISSUE THREAD

**Issue by dyc3** [biomejs/biome#9013]
### Environment information

<details>

```bash

CLI:
  Version:                      0.0.0
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           linux

Environment:
  BIOME_LOG_PATH:                    unset
  BIOME_LOG_PREFIX_NAME:             unset
  BIOME_LOG_LEVEL:                   unset
  BIOME_LOG_KIND:                    unset
  BIOME_CONFIG_PATH:                 unset
  BIOME_THREADS:                     unset
  BIOME_WATCHER_KIND:                unset
  BIOME_WATCHER_POLLING_INTERVAL:    unset
  NO_COLOR:                     unset
  TERM:                         xterm-256color
  JS_RUNTIME_VERSION:           unset
  JS_RUNTIME_NAME:              unset
  NODE_PACKAGE_MANAGER:         unset

Biome Configuration:
  Error:                        The configuration schema version does not match the CLI version 0.0.0
  Status:                       Loaded with errors
  Path:                         biome.json
  Formatter enabled:            false
  Linter enabled:               true
  Assist enabled:               false
  VCS enabled:                  true
  HTML full support enabled:    true

Workspace:
  Open Documents:               0

```

</details>

### What happened?

This is a bug to be fixed on `next`.

https://next.biomejs.dev/playground/?code=PABzAGMAcgBpAHAAdAAgAHMAZQB0AHUAcAA%2BAAoAIAAgAGMAbwBuAHMAdAAgAHMAdQBwAHAAbwByAHQAZQBkACAAPQAgAHIAZQBmACgAdAByAHUAZQApADsACgAgACAAYwBvAG4AcwB0ACAAZQBuAGEAYgBsAGUAZAAgAD0AIAByAGUAZgAoAGYAYQBsAHMAZQApADsACgAgACAAZgB1AG4AYwB0AGkAbwBuACAAdABvAGcAZwBsAGUAQwBhAHAAdABpAG8AbgBzACgAKQAgAHsACgAJAGUAbgBhAGIAbABlAGQALgB2AGEAbAB1AGUAIAA9ACAAIQBlAG4AYQBiAGwAZQBkAC4AdgBhAGwAdQBlADsACgAgACAAfQAKADwALwBzAGMAcgBpAHAAdAA%2BAAoACgA8AGIAdQB0AHQAbwBuAAoAIAAgAHQAeQBwAGUAPQAiAGIAdQB0AHQAbwBuACIACgAgACAAOgBkAGkAcwBhAGIAbABlAGQAPQAiACEAcwB1AHAAcABvAHIAdABlAGQAIgAKACAAIABAAGMAbABpAGMAawA9ACIAdABvAGcAZwBsAGUAQwBhAHAAdABpAG8AbgBzACgAKQAiAAoAPgAKACAAIABjAGwAaQBjAGsACgA8AC8AYgB1AHQAdABvAG4APgA%3D&language=vue&ruleDomains.react=recommended

Currently, expressions in Vue v-on directives (both longhand `v-on:` and shorthand `@`) don't mark any of the variables they use as used. 

In Vue, the value of a v-on directive is parsed as the body of an arrow function with no `{}` braces around it. My IDE decorates it like this:

<img width="276" height="58" alt="Image" src="https://github.com/user-attachments/assets/726ccf6d-8280-483e-bdd2-1ac09868a435" />

### Expected result

`noUnusedVariables` should not trigger here.

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct
